libr/core/file.c

```diff
@@ -486,7 +486,7 @@ R_API int r_core_bin_load(RCore *r, const char *filenameuri, ut64 baddr) {
 	}
 
 	if (cf) {
-		if ((filenameuri == NULL || !*filenameuri)) {
+		if (!(filenameuri || !*filenameuri)) {
```

Comment:
this is wrong. you are negating the whole conditional instead of just the value in filenameuri